swift/apple/FirezoneKit/Sources/FirezoneKit/AuthClient/AuthClient.swift

```diff
@@ -58,9 +66,26 @@ private final class WebAuthenticationSession: NSObject,
         }
 
         guard
-          let token = URLComponents(url: callbackURL, resolvingAgainstBaseURL: false)?
+          let stateInCallback = URLComponents(url: callbackURL, resolvingAgainstBaseURL: false)?
             .queryItems?
-            .first(where: { $0.name == "client_auth_token" })?
+            .first(where: { $0.name == "state" })?
+            .value
+        else {
+          continuation.resume(throwing: AuthClientError.invalidCallbackURL(callbackURL))
+          return
+        }
+
+        guard statePassedToPortal == stateInCallback else {
```

Comment:
All string comparisons involving secrets should use a constant-time comparison algorithm to protect against timing attacks.

Reply:
Attempted a constant time implementation in f881bc9. I think this won't get optimised out by the compiler -- I haven't checked the generated object code.